tools/preflight-checks.sh

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# Will be set to false if any of the steps fail
+did_checks_pass=true
+
+# Minimum versions
+MIN_GO_VERSION="1.17"
+MIN_NODE_VERSION="14.0.0"
+MIN_YARN_VERSION="1.22.11"
+
+# param 1 - required version
+# param 2 - current version
+# returns true or false if the version is ok
+is_version_ok() {
+  required_version=$1
+  current_version=$2
+  if [ "$(printf '%s\n' "$required_version" "$current_version" | sort -V | head -n1)" = "$required_version" ]; then 
+    return 0
+  else
+    return 1
+  fi
+}
+
+os() {
+  # If we're on OSX lets check for brew and coreutils as they are requirements
```

Comment:
Should we update the docs to mention running this script to see if your machine is compatible?

Reply:
Yeah sure